units/sleep_scheduler.go

```diff
@@ -86,55 +91,73 @@ func (j *sleepSchedulerJob) fixMissingNextScheduleTimes(ctx context.Context) err
-		if utility.IsZeroTime(h.SleepSchedule.NextStartTime) {
+		if utility.IsZeroTime(h.SleepSchedule.NextStartTime) && utility.IsZeroTime(h.SleepSchedule.NextStopTime) {
+			oldNextStart := h.SleepSchedule.NextStartTime
+			oldNextStop := h.SleepSchedule.NextStopTime
+			nextStart, nextStop, err := h.SleepSchedule.GetNextScheduledStartAndStopTimes(j.startedAt)
+			if err != nil {
+				catcher.Wrapf(err, "getting next start and stop times for host '%s'", h.Id)
+				continue
+			}
+			if err := h.SetNextScheduledStartAndStopTimes(ctx, nextStart, nextStop); err != nil {
+				catcher.Wrapf(err, "setting next start and stop times for host '%s'", h.Id)
+				continue
+			}
+			j.logMissingNextStart(h, oldNextStart, nextStart)
+			j.logMissingNextStop(h, oldNextStop, nextStop)
+		} else if utility.IsZeroTime(h.SleepSchedule.NextStartTime) {
 			oldNextStart := h.SleepSchedule.NextStartTime
-			nextStart, err := h.SleepSchedule.GetNextScheduledStartTime(now)
+			nextStart, err := h.SleepSchedule.GetNextScheduledStartTime(j.startedAt)
 			if err != nil {
 				catcher.Wrapf(err, "getting next start time for host '%s'", h.Id)
 				continue
 			}
-			if err := h.SetNextScheduledStart(ctx, nextStart); err != nil {
+			if err := h.SetNextScheduledStartTime(ctx, nextStart); err != nil {
 				catcher.Wrapf(err, "setting next start time for host '%s'", h.Id)
 				continue
 			}
-
-			grip.Notice(message.Fields{
-				"message":             "host is missing next start time, re-scheduled to next available start time",
-				"host_id":             h.Id,
-				"started_by":          h.StartedBy,
-				"old_next_start_time": oldNextStart,
-				"new_next_start_time": nextStart,
-				"job":                 j.ID(),
-			})
-		}
-		if utility.IsZeroTime(h.SleepSchedule.NextStopTime) {
+			j.logMissingNextStart(h, oldNextStart, nextStart)
+		} else if utility.IsZeroTime(h.SleepSchedule.NextStopTime) {
 			oldNextStop := h.SleepSchedule.NextStopTime
-			nextStop, err := h.SleepSchedule.GetNextScheduledStopTime(now)
+			nextStop, err := h.SleepSchedule.GetNextScheduledStopTime(j.startedAt)
 			if err != nil {
 				catcher.Wrapf(err, "getting next stop time for host '%s'", h.Id)
 				continue
 			}
-			if err := h.SetNextScheduledStop(ctx, nextStop); err != nil {
+			if err := h.SetNextScheduledStopTime(ctx, nextStop); err != nil {
 				catcher.Wrapf(err, "setting next stop time for host '%s'", h.Id)
 				continue
 			}
-
-			grip.Notice(message.Fields{
-				"message":            "host is missing next stop time, re-scheduled to next available stop time",
-				"host_id":            h.Id,
-				"started_by":         h.StartedBy,
-				"old_next_stop_time": oldNextStop,
-				"new_next_stop_time": nextStop,
-				"job":                j.ID(),
-			})
+			j.logMissingNextStop(h, oldNextStop, nextStop)
 		}
 	}
 	return catcher.Resolve()
 }
 
+func (j *sleepSchedulerJob) logMissingNextStart(h host.Host, oldNextStart, newNextStart time.Time) {
+	grip.Notice(message.Fields{
```

Comment:
nit: the grip level should be consistent with the `logExceededNextStopTimeout` func

Reply:
This one is a lower level on purpose. This one isn't actually a definite indicator that something is wrong, because it could be missing for normal reasons (like a temporary exemption ending). In contrast, a timeout means the sleep schedule wasn't able to stop/start the host for 30 minutes, which seems like it shouldn't happen if it's working correctly.